plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -646,38 +658,36 @@ def _get_pkglists_to_publish(self, erratum_unit):
         :rtype: list of dicts
         """
         pkglists_to_publish = []
-
-        # If we can pull a repo_id off the conduit, use that to generate repo-specific nevra
-        if self.context.conduit and hasattr(self.context.conduit, 'repo_id'):
-            repo_unit_nevra = self.context.get_repo_unit_nevra(erratum_unit)
-        else:
-            repo_unit_nevra = None
+        repo_id = self.get_repo().id
+        repo_unit_nevra = self._get_repo_unit_nevra(repo_id)
 
         seen_pkglists = set()
         for pkglist in erratum_unit.pkglist:
-            packages = tuple(sorted(p['filename'] for p in pkglist['packages']))
-            if packages in seen_pkglists:
-                continue
-            seen_pkglists.add(packages)
-
-            if repo_unit_nevra is None:
-                pkglists_to_publish.append(pkglist)
-                continue
+            # Due to a quirk in how yum merges pkglists in errata that appear in multiple
+            # repos, the pkglist collection name needs to be unique per-repo when published.
+            # repo_id is already unique and presumably validated, so use that.
+            pkglist['name'] = repo_id
```

Comment:
I have not tested it but do you know if it is ok to have multiple collections/pkglists in the erratum with the same name?
Just to illustrate what I mean (I wonder if they are all the same for the yum):

- multiple collections
```
<pkglist>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...1...>...</package>
    ...
  </collection>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...2...>...</package>
    ...
  </collection>
</pkglist>
``` 

- multiple pkglists
```
<pkglist>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...1...>...</package>
    ...
  </collection>
</pkglist>
<pkglist>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...2...>...</package>
    ...
  </collection>
</pkglist>
``` 

- multiple pkglists and collections
```
<pkglist>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...1...>...</package>
    ...
  </collection>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...2...>...</package>
    ...
  </collection>
</pkglist>
<pkglist>
  <collection short="smth">
    <name>my_repo_id</name>
    <package ...3...>...</package>
    ...
  </collection>
</pkglist>
``` 
Also I am not sure how this name is used except the merge logic and if it is shown somewhere, probably not.